In [ ]:
! pip install chembl_webresource_client

In [2]:
# Import necessary libraries
import pandas as pd
from chembl_webresource_client.new_client import new_client

In [3]:
#convert pubchemID to ChemBL ID
df_fla=pd.read_csv("fdb_molecules_22Nov.csv")
df_fla.head()

pubchem_id  molecular_weight  hbd_count  hba_count  xlogp  natural
0           4            75.111          2          2   -1.0        1
1          40           164.157          4          5   -1.6        0
2          47           130.143          1          3    1.1        0
3          49           116.116          1          3    0.7        1
4          51           146.098          2          5   -0.9        0

In [4]:
df_fla.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 25595 entries, 0 to 25594
Data columns (total 6 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   pubchem_id        25595 non-null  int64  
 1   molecular_weight  25595 non-null  float64
 2   hbd_count         25595 non-null  int64  
 3   hba_count         25595 non-null  int64  
 4   xlogp             22932 non-null  float64
 5   natural           25595 non-null  int64  
dtypes: float64(2), int64(4)
memory usage: 1.2 MB


In [6]:
df_nat=df_fla.loc[df_fla['natural'] == 1]
df_nat.reset_index(drop=True, inplace=True)
df_nat.info()
df_nat.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2254 entries, 0 to 2253
Data columns (total 6 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   pubchem_id        2254 non-null   int64  
 1   molecular_weight  2254 non-null   float64
 2   hbd_count         2254 non-null   int64  
 3   hba_count         2254 non-null   int64  
 4   xlogp             2203 non-null   float64
 5   natural           2254 non-null   int64  
dtypes: float64(2), int64(4)
memory usage: 105.8 KB


pubchem_id  molecular_weight  hbd_count  hba_count  xlogp  natural
0           4            75.111          2          2   -1.0        1
1          49           116.116          1          3    0.7        1
2          58           102.089          1          3    0.1        1
3          70           130.143          1          3    0.9        1
4          72           154.121          3          4    1.1        1

<font size=4> <b> Total fdb molecules were 25595 , natural molecules were only 2254 after conversion to chembl ids using Pubchem Identifier Exchange Service- Filename-"pubchem_tochembl.csv" . Only 1145 ids were left </b> </font>


In [10]:
df_nat.to_csv('fdb_natural_2254.csv', index=False)

In [ ]:
#####comverted pubchem is to chembl ids#### 

In [11]:
df_chem=pd.read_csv("pubchem_tochembl.csv")
df_chem.head()

pubchem_id     chembl_id
0           4  CHEMBL326602
1          49  CHEMBL146554
2          58  CHEMBL171246
3          70  CHEMBL445647
4          72   CHEMBL37537

In [12]:
df_chem.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2260 entries, 0 to 2259
Data columns (total 2 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   pubchem_id  2260 non-null   int64 
 1   chembl_id   1145 non-null   object
dtypes: int64(1), object(1)
memory usage: 35.4+ KB


In [13]:
df_chem_nona=df_chem[df_chem.chembl_id.notna()]
df_chem_nona.reset_index(drop=True, inplace=True)
#df_chem_nona.chembl_id.str.strip()
df_chem_nona.info()
df_chem_nona.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1145 entries, 0 to 1144
Data columns (total 2 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   pubchem_id  1145 non-null   int64 
 1   chembl_id   1145 non-null   object
dtypes: int64(1), object(1)
memory usage: 18.0+ KB


pubchem_id     chembl_id
0           4  CHEMBL326602
1          49  CHEMBL146554
2          58  CHEMBL171246
3          70  CHEMBL445647
4          72   CHEMBL37537

In [14]:
merge_dfchem_pub=df_nat.merge(df_chem_nona, on ="pubchem_id", how ="inner")
merge_dfchem_pub.info()
merge_dfchem_pub.head()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 1145 entries, 0 to 1144
Data columns (total 7 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   pubchem_id        1145 non-null   int64  
 1   molecular_weight  1145 non-null   float64
 2   hbd_count         1145 non-null   int64  
 3   hba_count         1145 non-null   int64  
 4   xlogp             1126 non-null   float64
 5   natural           1145 non-null   int64  
 6   chembl_id         1145 non-null   object 
dtypes: float64(2), int64(4), object(1)
memory usage: 71.6+ KB


pubchem_id  molecular_weight  hbd_count  hba_count  xlogp  natural  \
0           4            75.111          2          2   -1.0        1   
1          49           116.116          1          3    0.7        1   
2          58           102.089          1          3    0.1        1   
3          70           130.143          1          3    0.9        1   
4          72           154.121          3          4    1.1        1   

      chembl_id  
0  CHEMBL326602  
1  CHEMBL146554  
2  CHEMBL171246  
3  CHEMBL445647  
4   CHEMBL37537

In [10]:
molecules=list(merge_dfchem_pub['chembl_id'])
df2=merge_dfchem_pub
df2.to_csv('df2.csv', index= False)

In [11]:
import pandas as pd # uses pandas python module to view and analyse data
import requests # this is used to access json files

#====================================================================#

# using a list of natural fdb chembl ids bioactivity data scraping, Limit=5 records for each CHembl Id were scraped with Standard_type- IC50
# standard_type IC50 was selected , since it is most widely used for bioactivity prediction.

def find_bioactivities_for_molecules(molecules):
    
    limit = 5 # limit the number of records pulled back for each url call xxx
    standard_type='IC50'
    url_stem = "https://www.ebi.ac.uk" #This is the stem of the url
    url_full_string = url_stem + "/chembl/api/data/activity.json?molecule_chembl_id__in={}&limit={}&standard_type={}".format(molecules,limit,standard_type) #This is the full url with the specified input parameters
    url_full = requests.get( url_full_string ).json() #This calls the information back from the API using the 'requests' module, and converts it to json format
    url_activities = url_full['activities'] #This is a list of the results for activities

    # This 'while' loop iterates over several pages of records (if required), and collates the list of results
    while url_full['page_meta']['next']:
        url_full = requests.get(url_stem + url_full['page_meta']['next']).json()
        url_activities = url_activities + url_full['activities'] #Add result (as a list) to previous list of results

    # Convert the list of results into a Pandas dataframe:
    act_df = pd.DataFrame(url_activities)

    # Print out some useful information:
    print("This is the url string that calls the 'Activities' API with the initial query specification:\n{}".format(url_full_string) )
    print("\nThese are the available columns for the Activities API:\n{}".format(act_df.columns))

    #Specify which columns to keep so that the size of the dataframe becomes more manageable:
    act_df = act_df[[  'target_chembl_id','target_organism', 'target_pref_name'
                     ,'molecule_chembl_id','molecule_pref_name'
                     , 'pchembl_value', 'standard_type','standard_relation', 'standard_value', 'standard_units'
                     , 'assay_chembl_id']]

    return act_df

In [12]:
dic={}
df=pd.DataFrame()
id_list=[]
for i in range(len(df2)):
    k=df2.chembl_id[i]
    #dic["%s"%(k)]=find_bioactivities_for_molecules(k)
    try:
        data=find_bioactivities_for_molecules(k)
        df=df.append(data)
    #except KeyError as i:
    #print('Key Not Found in Employee Dictionary:',i)
    except KeyError:
        continue
    #id_list.append(dic["%s"%(i)])
#print(len(id_list))    

This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL326602&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index(['activity_comment', 'activity_id', 'activity_properties',
       'assay_chembl_id', 'assay_description', 'assay_type', 'bao_endpoint',
       'bao_format', 'bao_label', 'canonical_smiles', 'data_validity_comment',
       'data_validity_description', 'document_chembl_id', 'document_journal',
       'document_year', 'ligand_efficiency', 'molecule_chembl_id',
       'molecule_pref_name', 'parent_molecule_chembl_id', 'pchembl_value',
       'potential_duplicate', 'qudt_units', 'record_id', 'relation', 'src_id',
       'standard_flag', 'standard_relation', 'standard_text_value',
       'standard_type', 'standard_units', 'standard_upper_value',
       'standard_value', 'target_chembl_id', 'target_organism',
       'target_pr

This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL539&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index(['activity_comment', 'activity_id', 'activity_properties',
       'assay_chembl_id', 'assay_description', 'assay_type', 'bao_endpoint',
       'bao_format', 'bao_label', 'canonical_smiles', 'data_validity_comment',
       'data_validity_description', 'document_chembl_id', 'document_journal',
       'document_year', 'ligand_efficiency', 'molecule_chembl_id',
       'molecule_pref_name', 'parent_molecule_chembl_id', 'pchembl_value',
       'potential_duplicate', 'qudt_units', 'record_id', 'relation', 'src_id',
       'standard_flag', 'standard_relation', 'standard_text_value',
       'standard_type', 'standard_units', 'standard_upper_value',
       'standard_value', 'target_chembl_id', 'target_organism',
       'target_pref_

This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL1182&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index(['activity_comment', 'activity_id', 'activity_properties',
       'assay_chembl_id', 'assay_description', 'assay_type', 'bao_endpoint',
       'bao_format', 'bao_label', 'canonical_smiles', 'data_validity_comment',
       'data_validity_description', 'document_chembl_id', 'document_journal',
       'document_year', 'ligand_efficiency', 'molecule_chembl_id',
       'molecule_pref_name', 'parent_molecule_chembl_id', 'pchembl_value',
       'potential_duplicate', 'qudt_units', 'record_id', 'relation', 'src_id',
       'standard_flag', 'standard_relation', 'standard_text_value',
       'standard_type', 'standard_units', 'standard_upper_value',
       'standard_value', 'target_chembl_id', 'target_organism',
       'target_pref

This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL1232595&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index([], dtype='object')
This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL46931&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index(['activity_comment', 'activity_id', 'activity_properties',
       'assay_chembl_id', 'assay_description', 'assay_type', 'bao_endpoint',
       'bao_format', 'bao_label', 'canonical_smiles', 'data_validity_comment',
       'data_validity_description', 'document_chembl_id', 'document_journal',
       'document_year', 'ligand_efficiency', 'molecule_chembl_id',
       'molecule_pref_name', 'parent_molecule_chembl_id', 'pchembl_value',
   

This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL275626&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index([], dtype='object')
This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL12198&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index([], dtype='object')
This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL1200559&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index(['activity_comment', 'activity_id', 'activity_properties',
       'assay_chembl_id', 'assay_description', 'assay_type', 'bao_endpoint',
       'bao

This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL692&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index([], dtype='object')
This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL183419&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index([], dtype='object')
This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL71595&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index(['activity_comment', 'activity_id', 'activity_properties',
       'assay_chembl_id', 'assay_description', 'assay_type', 'bao_endpoint',
       'bao_for

This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL26215&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index([], dtype='object')
This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL146755&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index(['activity_comment', 'activity_id', 'activity_properties',
       'assay_chembl_id', 'assay_description', 'assay_type', 'bao_endpoint',
       'bao_format', 'bao_label', 'canonical_smiles', 'data_validity_comment',
       'data_validity_description', 'document_chembl_id', 'document_journal',
       'document_year', 'ligand_efficiency', 'molecule_chembl_id',
       'molecule_pref_name', 'parent_molecule_chembl_id', 'pchembl_value',
    

This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL14687&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index([], dtype='object')
This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL14021&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index(['activity_comment', 'activity_id', 'activity_properties',
       'assay_chembl_id', 'assay_description', 'assay_type', 'bao_endpoint',
       'bao_format', 'bao_label', 'canonical_smiles', 'data_validity_comment',
       'data_validity_description', 'document_chembl_id', 'document_journal',
       'document_year', 'ligand_efficiency', 'molecule_chembl_id',
       'molecule_pref_name', 'parent_molecule_chembl_id', 'pchembl_value',
     

This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL1236482&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index([], dtype='object')
This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL9113&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index(['activity_comment', 'activity_id', 'activity_properties',
       'assay_chembl_id', 'assay_description', 'assay_type', 'bao_endpoint',
       'bao_format', 'bao_label', 'canonical_smiles', 'data_validity_comment',
       'data_validity_description', 'document_chembl_id', 'document_journal',
       'document_year', 'ligand_efficiency', 'molecule_chembl_id',
       'molecule_pref_name', 'parent_molecule_chembl_id', 'pchembl_value',
    

This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL93353&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index(['activity_comment', 'activity_id', 'activity_properties',
       'assay_chembl_id', 'assay_description', 'assay_type', 'bao_endpoint',
       'bao_format', 'bao_label', 'canonical_smiles', 'data_validity_comment',
       'data_validity_description', 'document_chembl_id', 'document_journal',
       'document_year', 'ligand_efficiency', 'molecule_chembl_id',
       'molecule_pref_name', 'parent_molecule_chembl_id', 'pchembl_value',
       'potential_duplicate', 'qudt_units', 'record_id', 'relation', 'src_id',
       'standard_flag', 'standard_relation', 'standard_text_value',
       'standard_type', 'standard_units', 'standard_upper_value',
       'standard_value', 'target_chembl_id', 'target_organism',
       'target_pre

This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL107498&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index(['activity_comment', 'activity_id', 'activity_properties',
       'assay_chembl_id', 'assay_description', 'assay_type', 'bao_endpoint',
       'bao_format', 'bao_label', 'canonical_smiles', 'data_validity_comment',
       'data_validity_description', 'document_chembl_id', 'document_journal',
       'document_year', 'ligand_efficiency', 'molecule_chembl_id',
       'molecule_pref_name', 'parent_molecule_chembl_id', 'pchembl_value',
       'potential_duplicate', 'qudt_units', 'record_id', 'relation', 'src_id',
       'standard_flag', 'standard_relation', 'standard_text_value',
       'standard_type', 'standard_units', 'standard_upper_value',
       'standard_value', 'target_chembl_id', 'target_organism',
       'target_pr

This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL108766&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index(['activity_comment', 'activity_id', 'activity_properties',
       'assay_chembl_id', 'assay_description', 'assay_type', 'bao_endpoint',
       'bao_format', 'bao_label', 'canonical_smiles', 'data_validity_comment',
       'data_validity_description', 'document_chembl_id', 'document_journal',
       'document_year', 'ligand_efficiency', 'molecule_chembl_id',
       'molecule_pref_name', 'parent_molecule_chembl_id', 'pchembl_value',
       'potential_duplicate', 'qudt_units', 'record_id', 'relation', 'src_id',
       'standard_flag', 'standard_relation', 'standard_text_value',
       'standard_type', 'standard_units', 'standard_upper_value',
       'standard_value', 'target_chembl_id', 'target_organism',
       'target_pr

This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL7983&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index(['activity_comment', 'activity_id', 'activity_properties',
       'assay_chembl_id', 'assay_description', 'assay_type', 'bao_endpoint',
       'bao_format', 'bao_label', 'canonical_smiles', 'data_validity_comment',
       'data_validity_description', 'document_chembl_id', 'document_journal',
       'document_year', 'ligand_efficiency', 'molecule_chembl_id',
       'molecule_pref_name', 'parent_molecule_chembl_id', 'pchembl_value',
       'potential_duplicate', 'qudt_units', 'record_id', 'relation', 'src_id',
       'standard_flag', 'standard_relation', 'standard_text_value',
       'standard_type', 'standard_units', 'standard_upper_value',
       'standard_value', 'target_chembl_id', 'target_organism',
       'target_pref

This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL1682&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index([], dtype='object')
This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL1222250&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index(['activity_comment', 'activity_id', 'activity_properties',
       'assay_chembl_id', 'assay_description', 'assay_type', 'bao_endpoint',
       'bao_format', 'bao_label', 'canonical_smiles', 'data_validity_comment',
       'data_validity_description', 'document_chembl_id', 'document_journal',
       'document_year', 'ligand_efficiency', 'molecule_chembl_id',
       'molecule_pref_name', 'parent_molecule_chembl_id', 'pchembl_value',
    

This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL253582&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index(['activity_comment', 'activity_id', 'activity_properties',
       'assay_chembl_id', 'assay_description', 'assay_type', 'bao_endpoint',
       'bao_format', 'bao_label', 'canonical_smiles', 'data_validity_comment',
       'data_validity_description', 'document_chembl_id', 'document_journal',
       'document_year', 'ligand_efficiency', 'molecule_chembl_id',
       'molecule_pref_name', 'parent_molecule_chembl_id', 'pchembl_value',
       'potential_duplicate', 'qudt_units', 'record_id', 'relation', 'src_id',
       'standard_flag', 'standard_relation', 'standard_text_value',
       'standard_type', 'standard_units', 'standard_upper_value',
       'standard_value', 'target_chembl_id', 'target_organism',
       'target_pr

This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL107&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index(['activity_comment', 'activity_id', 'activity_properties',
       'assay_chembl_id', 'assay_description', 'assay_type', 'bao_endpoint',
       'bao_format', 'bao_label', 'canonical_smiles', 'data_validity_comment',
       'data_validity_description', 'document_chembl_id', 'document_journal',
       'document_year', 'ligand_efficiency', 'molecule_chembl_id',
       'molecule_pref_name', 'parent_molecule_chembl_id', 'pchembl_value',
       'potential_duplicate', 'qudt_units', 'record_id', 'relation', 'src_id',
       'standard_flag', 'standard_relation', 'standard_text_value',
       'standard_type', 'standard_units', 'standard_upper_value',
       'standard_value', 'target_chembl_id', 'target_organism',
       'target_pref_

This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL14449&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index([], dtype='object')
This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL45967&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index(['activity_comment', 'activity_id', 'activity_properties',
       'assay_chembl_id', 'assay_description', 'assay_type', 'bao_endpoint',
       'bao_format', 'bao_label', 'canonical_smiles', 'data_validity_comment',
       'data_validity_description', 'document_chembl_id', 'document_journal',
       'document_year', 'ligand_efficiency', 'molecule_chembl_id',
       'molecule_pref_name', 'parent_molecule_chembl_id', 'pchembl_value',
     

This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL1651998&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index([], dtype='object')
This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL1414114&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index([], dtype='object')
This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL1513871&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index([], dtype='object')
This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/ap

This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL1719455&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index([], dtype='object')
This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL2252746&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index([], dtype='object')
This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL1898498&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index([], dtype='object')
This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/ap

This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL108861&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index(['activity_comment', 'activity_id', 'activity_properties',
       'assay_chembl_id', 'assay_description', 'assay_type', 'bao_endpoint',
       'bao_format', 'bao_label', 'canonical_smiles', 'data_validity_comment',
       'data_validity_description', 'document_chembl_id', 'document_journal',
       'document_year', 'ligand_efficiency', 'molecule_chembl_id',
       'molecule_pref_name', 'parent_molecule_chembl_id', 'pchembl_value',
       'potential_duplicate', 'qudt_units', 'record_id', 'relation', 'src_id',
       'standard_flag', 'standard_relation', 'standard_text_value',
       'standard_type', 'standard_units', 'standard_upper_value',
       'standard_value', 'target_chembl_id', 'target_organism',
       'target_pr

This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL45005&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index(['activity_comment', 'activity_id', 'activity_properties',
       'assay_chembl_id', 'assay_description', 'assay_type', 'bao_endpoint',
       'bao_format', 'bao_label', 'canonical_smiles', 'data_validity_comment',
       'data_validity_description', 'document_chembl_id', 'document_journal',
       'document_year', 'ligand_efficiency', 'molecule_chembl_id',
       'molecule_pref_name', 'parent_molecule_chembl_id', 'pchembl_value',
       'potential_duplicate', 'qudt_units', 'record_id', 'relation', 'src_id',
       'standard_flag', 'standard_relation', 'standard_text_value',
       'standard_type', 'standard_units', 'standard_upper_value',
       'standard_value', 'target_chembl_id', 'target_organism',
       'target_pre

This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL3560314&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index([], dtype='object')
This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL328441&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index(['activity_comment', 'activity_id', 'activity_properties',
       'assay_chembl_id', 'assay_description', 'assay_type', 'bao_endpoint',
       'bao_format', 'bao_label', 'canonical_smiles', 'data_validity_comment',
       'data_validity_description', 'document_chembl_id', 'document_journal',
       'document_year', 'ligand_efficiency', 'molecule_chembl_id',
       'molecule_pref_name', 'parent_molecule_chembl_id', 'pchembl_value',
  

This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL201083&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index(['activity_comment', 'activity_id', 'activity_properties',
       'assay_chembl_id', 'assay_description', 'assay_type', 'bao_endpoint',
       'bao_format', 'bao_label', 'canonical_smiles', 'data_validity_comment',
       'data_validity_description', 'document_chembl_id', 'document_journal',
       'document_year', 'ligand_efficiency', 'molecule_chembl_id',
       'molecule_pref_name', 'parent_molecule_chembl_id', 'pchembl_value',
       'potential_duplicate', 'qudt_units', 'record_id', 'relation', 'src_id',
       'standard_flag', 'standard_relation', 'standard_text_value',
       'standard_type', 'standard_units', 'standard_upper_value',
       'standard_value', 'target_chembl_id', 'target_organism',
       'target_pr

This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL3184059&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index([], dtype='object')
This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL3184025&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index([], dtype='object')
This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL3184929&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index([], dtype='object')
This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/ap

This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL294431&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index(['activity_comment', 'activity_id', 'activity_properties',
       'assay_chembl_id', 'assay_description', 'assay_type', 'bao_endpoint',
       'bao_format', 'bao_label', 'canonical_smiles', 'data_validity_comment',
       'data_validity_description', 'document_chembl_id', 'document_journal',
       'document_year', 'ligand_efficiency', 'molecule_chembl_id',
       'molecule_pref_name', 'parent_molecule_chembl_id', 'pchembl_value',
       'potential_duplicate', 'qudt_units', 'record_id', 'relation', 'src_id',
       'standard_flag', 'standard_relation', 'standard_text_value',
       'standard_type', 'standard_units', 'standard_upper_value',
       'standard_value', 'target_chembl_id', 'target_organism',
       'target_pr

This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL556489&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index([], dtype='object')
This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL537970&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index([], dtype='object')
This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL721&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index(['activity_comment', 'activity_id', 'activity_properties',
       'assay_chembl_id', 'assay_description', 'assay_type', 'bao_endpoint',
       'bao_fo

This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL22976&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index([], dtype='object')
This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL1986332&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index([], dtype='object')
This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL3183097&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index([], dtype='object')
This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/

This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL3184829&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index([], dtype='object')
This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL2137647&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index([], dtype='object')
This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL18893&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index(['activity_comment', 'activity_id', 'activity_properties',
       'assay_chembl_id', 'assay_description', 'assay_type', 'bao_endpoint',
       'ba

This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL1591973&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index(['activity_comment', 'activity_id', 'activity_properties',
       'assay_chembl_id', 'assay_description', 'assay_type', 'bao_endpoint',
       'bao_format', 'bao_label', 'canonical_smiles', 'data_validity_comment',
       'data_validity_description', 'document_chembl_id', 'document_journal',
       'document_year', 'ligand_efficiency', 'molecule_chembl_id',
       'molecule_pref_name', 'parent_molecule_chembl_id', 'pchembl_value',
       'potential_duplicate', 'qudt_units', 'record_id', 'relation', 'src_id',
       'standard_flag', 'standard_relation', 'standard_text_value',
       'standard_type', 'standard_units', 'standard_upper_value',
       'standard_value', 'target_chembl_id', 'target_organism',
       'target_p

This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL2228460&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index([], dtype='object')
This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL24832&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index([], dtype='object')
This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL24022&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index([], dtype='object')
This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/da

This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL15841&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index([], dtype='object')
This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL271663&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index(['activity_comment', 'activity_id', 'activity_properties',
       'assay_chembl_id', 'assay_description', 'assay_type', 'bao_endpoint',
       'bao_format', 'bao_label', 'canonical_smiles', 'data_validity_comment',
       'data_validity_description', 'document_chembl_id', 'document_journal',
       'document_year', 'ligand_efficiency', 'molecule_chembl_id',
       'molecule_pref_name', 'parent_molecule_chembl_id', 'pchembl_value',
    

This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL3183988&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index([], dtype='object')
This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL3184678&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index([], dtype='object')
This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL2104112&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index([], dtype='object')
This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/ap

This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL14184&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index(['activity_comment', 'activity_id', 'activity_properties',
       'assay_chembl_id', 'assay_description', 'assay_type', 'bao_endpoint',
       'bao_format', 'bao_label', 'canonical_smiles', 'data_validity_comment',
       'data_validity_description', 'document_chembl_id', 'document_journal',
       'document_year', 'ligand_efficiency', 'molecule_chembl_id',
       'molecule_pref_name', 'parent_molecule_chembl_id', 'pchembl_value',
       'potential_duplicate', 'qudt_units', 'record_id', 'relation', 'src_id',
       'standard_flag', 'standard_relation', 'standard_text_value',
       'standard_type', 'standard_units', 'standard_upper_value',
       'standard_value', 'target_chembl_id', 'target_organism',
       'target_pre

This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL311498&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index(['activity_comment', 'activity_id', 'activity_properties',
       'assay_chembl_id', 'assay_description', 'assay_type', 'bao_endpoint',
       'bao_format', 'bao_label', 'canonical_smiles', 'data_validity_comment',
       'data_validity_description', 'document_chembl_id', 'document_journal',
       'document_year', 'ligand_efficiency', 'molecule_chembl_id',
       'molecule_pref_name', 'parent_molecule_chembl_id', 'pchembl_value',
       'potential_duplicate', 'qudt_units', 'record_id', 'relation', 'src_id',
       'standard_flag', 'standard_relation', 'standard_text_value',
       'standard_type', 'standard_units', 'standard_upper_value',
       'standard_value', 'target_chembl_id', 'target_organism',
       'target_pr

This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL166223&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index([], dtype='object')
This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL4169727&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index([], dtype='object')
This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL281202&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index(['activity_comment', 'activity_id', 'activity_properties',
       'assay_chembl_id', 'assay_description', 'assay_type', 'bao_endpoint',
       'ba

This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL444711&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index([], dtype='object')
This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL23025&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index([], dtype='object')
This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL47483&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index([], dtype='object')
This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/dat

This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL452729&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index([], dtype='object')
This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL487998&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index([], dtype='object')
This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL3561586&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index([], dtype='object')
This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/

This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL443470&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index([], dtype='object')
This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL192809&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index([], dtype='object')
This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL2269086&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index([], dtype='object')
This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/

This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL26561&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index([], dtype='object')
This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL3185332&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index([], dtype='object')
This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL107874&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index(['activity_comment', 'activity_id', 'activity_properties',
       'assay_chembl_id', 'assay_description', 'assay_type', 'bao_endpoint',
       'bao

This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL3182189&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index([], dtype='object')
This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL365740&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index(['activity_comment', 'activity_id', 'activity_properties',
       'assay_chembl_id', 'assay_description', 'assay_type', 'bao_endpoint',
       'bao_format', 'bao_label', 'canonical_smiles', 'data_validity_comment',
       'data_validity_description', 'document_chembl_id', 'document_journal',
       'document_year', 'ligand_efficiency', 'molecule_chembl_id',
       'molecule_pref_name', 'parent_molecule_chembl_id', 'pchembl_value',
  

This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL3186281&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index([], dtype='object')
This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL132474&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index([], dtype='object')
This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL11945&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index([], dtype='object')
This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/d

This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL134537&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index([], dtype='object')
This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL2252095&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index([], dtype='object')
This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL123040&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index(['activity_comment', 'activity_id', 'activity_properties',
       'assay_chembl_id', 'assay_description', 'assay_type', 'bao_endpoint',
       'ba

This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL3182738&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index([], dtype='object')
This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL449810&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index([], dtype='object')
This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL3182794&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index([], dtype='object')
This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api

This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL105912&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index(['activity_comment', 'activity_id', 'activity_properties',
       'assay_chembl_id', 'assay_description', 'assay_type', 'bao_endpoint',
       'bao_format', 'bao_label', 'canonical_smiles', 'data_validity_comment',
       'data_validity_description', 'document_chembl_id', 'document_journal',
       'document_year', 'ligand_efficiency', 'molecule_chembl_id',
       'molecule_pref_name', 'parent_molecule_chembl_id', 'pchembl_value',
       'potential_duplicate', 'qudt_units', 'record_id', 'relation', 'src_id',
       'standard_flag', 'standard_relation', 'standard_text_value',
       'standard_type', 'standard_units', 'standard_upper_value',
       'standard_value', 'target_chembl_id', 'target_organism',
       'target_pr

This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL320146&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index([], dtype='object')
This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL2271411&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index([], dtype='object')
This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL3187840&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index([], dtype='object')
This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api

This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL161598&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index(['activity_comment', 'activity_id', 'activity_properties',
       'assay_chembl_id', 'assay_description', 'assay_type', 'bao_endpoint',
       'bao_format', 'bao_label', 'canonical_smiles', 'data_validity_comment',
       'data_validity_description', 'document_chembl_id', 'document_journal',
       'document_year', 'ligand_efficiency', 'molecule_chembl_id',
       'molecule_pref_name', 'parent_molecule_chembl_id', 'pchembl_value',
       'potential_duplicate', 'qudt_units', 'record_id', 'relation', 'src_id',
       'standard_flag', 'standard_relation', 'standard_text_value',
       'standard_type', 'standard_units', 'standard_upper_value',
       'standard_value', 'target_chembl_id', 'target_organism',
       'target_pr

This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL328465&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index([], dtype='object')
This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL96852&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index([], dtype='object')
This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL3186540&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index([], dtype='object')
This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/d

This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL1374805&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index([], dtype='object')
This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL3560312&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index([], dtype='object')
This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL47155&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index([], dtype='object')
This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/

This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL3186269&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index([], dtype='object')
This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL1789171&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index([], dtype='object')
This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL3185445&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index([], dtype='object')
This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/ap

This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL125743&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index(['activity_comment', 'activity_id', 'activity_properties',
       'assay_chembl_id', 'assay_description', 'assay_type', 'bao_endpoint',
       'bao_format', 'bao_label', 'canonical_smiles', 'data_validity_comment',
       'data_validity_description', 'document_chembl_id', 'document_journal',
       'document_year', 'ligand_efficiency', 'molecule_chembl_id',
       'molecule_pref_name', 'parent_molecule_chembl_id', 'pchembl_value',
       'potential_duplicate', 'qudt_units', 'record_id', 'relation', 'src_id',
       'standard_flag', 'standard_relation', 'standard_text_value',
       'standard_type', 'standard_units', 'standard_upper_value',
       'standard_value', 'target_chembl_id', 'target_organism',
       'target_pr

This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL3593941&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index([], dtype='object')
This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL1079147&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index([], dtype='object')
This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL21824&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index([], dtype='object')
This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/

This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL379630&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index(['activity_comment', 'activity_id', 'activity_properties',
       'assay_chembl_id', 'assay_description', 'assay_type', 'bao_endpoint',
       'bao_format', 'bao_label', 'canonical_smiles', 'data_validity_comment',
       'data_validity_description', 'document_chembl_id', 'document_journal',
       'document_year', 'ligand_efficiency', 'molecule_chembl_id',
       'molecule_pref_name', 'parent_molecule_chembl_id', 'pchembl_value',
       'potential_duplicate', 'qudt_units', 'record_id', 'relation', 'src_id',
       'standard_flag', 'standard_relation', 'standard_text_value',
       'standard_type', 'standard_units', 'standard_upper_value',
       'standard_value', 'target_chembl_id', 'target_organism',
       'target_pr

This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL97593&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index([], dtype='object')
This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL3593942&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index([], dtype='object')
This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL298717&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index(['activity_comment', 'activity_id', 'activity_properties',
       'assay_chembl_id', 'assay_description', 'assay_type', 'bao_endpoint',
       'bao

This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL3593943&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index([], dtype='object')
This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL3186132&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index([], dtype='object')
This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL2004717&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index([], dtype='object')
This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/ap

This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL171281&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index([], dtype='object')
This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL1990994&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index([], dtype='object')
This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL1988732&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index([], dtype='object')
This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api

This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL36327&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index(['activity_comment', 'activity_id', 'activity_properties',
       'assay_chembl_id', 'assay_description', 'assay_type', 'bao_endpoint',
       'bao_format', 'bao_label', 'canonical_smiles', 'data_validity_comment',
       'data_validity_description', 'document_chembl_id', 'document_journal',
       'document_year', 'ligand_efficiency', 'molecule_chembl_id',
       'molecule_pref_name', 'parent_molecule_chembl_id', 'pchembl_value',
       'potential_duplicate', 'qudt_units', 'record_id', 'relation', 'src_id',
       'standard_flag', 'standard_relation', 'standard_text_value',
       'standard_type', 'standard_units', 'standard_upper_value',
       'standard_value', 'target_chembl_id', 'target_organism',
       'target_pre

This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL1661&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index(['activity_comment', 'activity_id', 'activity_properties',
       'assay_chembl_id', 'assay_description', 'assay_type', 'bao_endpoint',
       'bao_format', 'bao_label', 'canonical_smiles', 'data_validity_comment',
       'data_validity_description', 'document_chembl_id', 'document_journal',
       'document_year', 'ligand_efficiency', 'molecule_chembl_id',
       'molecule_pref_name', 'parent_molecule_chembl_id', 'pchembl_value',
       'potential_duplicate', 'qudt_units', 'record_id', 'relation', 'src_id',
       'standard_flag', 'standard_relation', 'standard_text_value',
       'standard_type', 'standard_units', 'standard_upper_value',
       'standard_value', 'target_chembl_id', 'target_organism',
       'target_pref

This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL221542&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index(['activity_comment', 'activity_id', 'activity_properties',
       'assay_chembl_id', 'assay_description', 'assay_type', 'bao_endpoint',
       'bao_format', 'bao_label', 'canonical_smiles', 'data_validity_comment',
       'data_validity_description', 'document_chembl_id', 'document_journal',
       'document_year', 'ligand_efficiency', 'molecule_chembl_id',
       'molecule_pref_name', 'parent_molecule_chembl_id', 'pchembl_value',
       'potential_duplicate', 'qudt_units', 'record_id', 'relation', 'src_id',
       'standard_flag', 'standard_relation', 'standard_text_value',
       'standard_type', 'standard_units', 'standard_upper_value',
       'standard_value', 'target_chembl_id', 'target_organism',
       'target_pr

This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL364713&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index(['activity_comment', 'activity_id', 'activity_properties',
       'assay_chembl_id', 'assay_description', 'assay_type', 'bao_endpoint',
       'bao_format', 'bao_label', 'canonical_smiles', 'data_validity_comment',
       'data_validity_description', 'document_chembl_id', 'document_journal',
       'document_year', 'ligand_efficiency', 'molecule_chembl_id',
       'molecule_pref_name', 'parent_molecule_chembl_id', 'pchembl_value',
       'potential_duplicate', 'qudt_units', 'record_id', 'relation', 'src_id',
       'standard_flag', 'standard_relation', 'standard_text_value',
       'standard_type', 'standard_units', 'standard_upper_value',
       'standard_value', 'target_chembl_id', 'target_organism',
       'target_pr

This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL66&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index(['activity_comment', 'activity_id', 'activity_properties',
       'assay_chembl_id', 'assay_description', 'assay_type', 'bao_endpoint',
       'bao_format', 'bao_label', 'canonical_smiles', 'data_validity_comment',
       'data_validity_description', 'document_chembl_id', 'document_journal',
       'document_year', 'ligand_efficiency', 'molecule_chembl_id',
       'molecule_pref_name', 'parent_molecule_chembl_id', 'pchembl_value',
       'potential_duplicate', 'qudt_units', 'record_id', 'relation', 'src_id',
       'standard_flag', 'standard_relation', 'standard_text_value',
       'standard_type', 'standard_units', 'standard_upper_value',
       'standard_value', 'target_chembl_id', 'target_organism',
       'target_pref_n

This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL1096927&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index(['activity_comment', 'activity_id', 'activity_properties',
       'assay_chembl_id', 'assay_description', 'assay_type', 'bao_endpoint',
       'bao_format', 'bao_label', 'canonical_smiles', 'data_validity_comment',
       'data_validity_description', 'document_chembl_id', 'document_journal',
       'document_year', 'ligand_efficiency', 'molecule_chembl_id',
       'molecule_pref_name', 'parent_molecule_chembl_id', 'pchembl_value',
       'potential_duplicate', 'qudt_units', 'record_id', 'relation', 'src_id',
       'standard_flag', 'standard_relation', 'standard_text_value',
       'standard_type', 'standard_units', 'standard_upper_value',
       'standard_value', 'target_chembl_id', 'target_organism',
       'target_p

This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL165&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index(['activity_comment', 'activity_id', 'activity_properties',
       'assay_chembl_id', 'assay_description', 'assay_type', 'bao_endpoint',
       'bao_format', 'bao_label', 'canonical_smiles', 'data_validity_comment',
       'data_validity_description', 'document_chembl_id', 'document_journal',
       'document_year', 'ligand_efficiency', 'molecule_chembl_id',
       'molecule_pref_name', 'parent_molecule_chembl_id', 'pchembl_value',
       'potential_duplicate', 'qudt_units', 'record_id', 'relation', 'src_id',
       'standard_flag', 'standard_relation', 'standard_text_value',
       'standard_type', 'standard_units', 'standard_upper_value',
       'standard_value', 'target_chembl_id', 'target_organism',
       'target_pref_

This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL3188345&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index([], dtype='object')
This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL117865&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index([], dtype='object')
This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL94514&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index([], dtype='object')
This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/d

This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL452630&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index(['activity_comment', 'activity_id', 'activity_properties',
       'assay_chembl_id', 'assay_description', 'assay_type', 'bao_endpoint',
       'bao_format', 'bao_label', 'canonical_smiles', 'data_validity_comment',
       'data_validity_description', 'document_chembl_id', 'document_journal',
       'document_year', 'ligand_efficiency', 'molecule_chembl_id',
       'molecule_pref_name', 'parent_molecule_chembl_id', 'pchembl_value',
       'potential_duplicate', 'qudt_units', 'record_id', 'relation', 'src_id',
       'standard_flag', 'standard_relation', 'standard_text_value',
       'standard_type', 'standard_units', 'standard_upper_value',
       'standard_value', 'target_chembl_id', 'target_organism',
       'target_pr

This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL43185&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index(['activity_comment', 'activity_id', 'activity_properties',
       'assay_chembl_id', 'assay_description', 'assay_type', 'bao_endpoint',
       'bao_format', 'bao_label', 'canonical_smiles', 'data_validity_comment',
       'data_validity_description', 'document_chembl_id', 'document_journal',
       'document_year', 'ligand_efficiency', 'molecule_chembl_id',
       'molecule_pref_name', 'parent_molecule_chembl_id', 'pchembl_value',
       'potential_duplicate', 'qudt_units', 'record_id', 'relation', 'src_id',
       'standard_flag', 'standard_relation', 'standard_text_value',
       'standard_type', 'standard_units', 'standard_upper_value',
       'standard_value', 'target_chembl_id', 'target_organism',
       'target_pre

This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL347285&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index(['activity_comment', 'activity_id', 'activity_properties',
       'assay_chembl_id', 'assay_description', 'assay_type', 'bao_endpoint',
       'bao_format', 'bao_label', 'canonical_smiles', 'data_validity_comment',
       'data_validity_description', 'document_chembl_id', 'document_journal',
       'document_year', 'ligand_efficiency', 'molecule_chembl_id',
       'molecule_pref_name', 'parent_molecule_chembl_id', 'pchembl_value',
       'potential_duplicate', 'qudt_units', 'record_id', 'relation', 'src_id',
       'standard_flag', 'standard_relation', 'standard_text_value',
       'standard_type', 'standard_units', 'standard_upper_value',
       'standard_value', 'target_chembl_id', 'target_organism',
       'target_pr

This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL2251602&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index([], dtype='object')
This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL2268549&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index(['activity_comment', 'activity_id', 'activity_properties',
       'assay_chembl_id', 'assay_description', 'assay_type', 'bao_endpoint',
       'bao_format', 'bao_label', 'canonical_smiles', 'data_validity_comment',
       'data_validity_description', 'document_chembl_id', 'document_journal',
       'document_year', 'ligand_efficiency', 'molecule_chembl_id',
       'molecule_pref_name', 'parent_molecule_chembl_id', 'pchembl_value',
 

This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL1873035&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index([], dtype='object')
This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL2251453&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index([], dtype='object')
This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL170&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index(['activity_comment', 'activity_id', 'activity_properties',
       'assay_chembl_id', 'assay_description', 'assay_type', 'bao_endpoint',
       'bao_

This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL1644111&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index(['activity_comment', 'activity_id', 'activity_properties',
       'assay_chembl_id', 'assay_description', 'assay_type', 'bao_endpoint',
       'bao_format', 'bao_label', 'canonical_smiles', 'data_validity_comment',
       'data_validity_description', 'document_chembl_id', 'document_journal',
       'document_year', 'ligand_efficiency', 'molecule_chembl_id',
       'molecule_pref_name', 'parent_molecule_chembl_id', 'pchembl_value',
       'potential_duplicate', 'qudt_units', 'record_id', 'relation', 'src_id',
       'standard_flag', 'standard_relation', 'standard_text_value',
       'standard_type', 'standard_units', 'standard_upper_value',
       'standard_value', 'target_chembl_id', 'target_organism',
       'target_p

This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL150&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index(['activity_comment', 'activity_id', 'activity_properties',
       'assay_chembl_id', 'assay_description', 'assay_type', 'bao_endpoint',
       'bao_format', 'bao_label', 'canonical_smiles', 'data_validity_comment',
       'data_validity_description', 'document_chembl_id', 'document_journal',
       'document_year', 'ligand_efficiency', 'molecule_chembl_id',
       'molecule_pref_name', 'parent_molecule_chembl_id', 'pchembl_value',
       'potential_duplicate', 'qudt_units', 'record_id', 'relation', 'src_id',
       'standard_flag', 'standard_relation', 'standard_text_value',
       'standard_type', 'standard_units', 'standard_upper_value',
       'standard_value', 'target_chembl_id', 'target_organism',
       'target_pref_

This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL251280&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index(['activity_comment', 'activity_id', 'activity_properties',
       'assay_chembl_id', 'assay_description', 'assay_type', 'bao_endpoint',
       'bao_format', 'bao_label', 'canonical_smiles', 'data_validity_comment',
       'data_validity_description', 'document_chembl_id', 'document_journal',
       'document_year', 'ligand_efficiency', 'molecule_chembl_id',
       'molecule_pref_name', 'parent_molecule_chembl_id', 'pchembl_value',
       'potential_duplicate', 'qudt_units', 'record_id', 'relation', 'src_id',
       'standard_flag', 'standard_relation', 'standard_text_value',
       'standard_type', 'standard_units', 'standard_upper_value',
       'standard_value', 'target_chembl_id', 'target_organism',
       'target_pr

This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL28626&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index(['activity_comment', 'activity_id', 'activity_properties',
       'assay_chembl_id', 'assay_description', 'assay_type', 'bao_endpoint',
       'bao_format', 'bao_label', 'canonical_smiles', 'data_validity_comment',
       'data_validity_description', 'document_chembl_id', 'document_journal',
       'document_year', 'ligand_efficiency', 'molecule_chembl_id',
       'molecule_pref_name', 'parent_molecule_chembl_id', 'pchembl_value',
       'potential_duplicate', 'qudt_units', 'record_id', 'relation', 'src_id',
       'standard_flag', 'standard_relation', 'standard_text_value',
       'standard_type', 'standard_units', 'standard_upper_value',
       'standard_value', 'target_chembl_id', 'target_organism',
       'target_pre

This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL472877&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index(['activity_comment', 'activity_id', 'activity_properties',
       'assay_chembl_id', 'assay_description', 'assay_type', 'bao_endpoint',
       'bao_format', 'bao_label', 'canonical_smiles', 'data_validity_comment',
       'data_validity_description', 'document_chembl_id', 'document_journal',
       'document_year', 'ligand_efficiency', 'molecule_chembl_id',
       'molecule_pref_name', 'parent_molecule_chembl_id', 'pchembl_value',
       'potential_duplicate', 'qudt_units', 'record_id', 'relation', 'src_id',
       'standard_flag', 'standard_relation', 'standard_text_value',
       'standard_type', 'standard_units', 'standard_upper_value',
       'standard_value', 'target_chembl_id', 'target_organism',
       'target_pr

This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL3184538&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index([], dtype='object')
This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL25424&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index(['activity_comment', 'activity_id', 'activity_properties',
       'assay_chembl_id', 'assay_description', 'assay_type', 'bao_endpoint',
       'bao_format', 'bao_label', 'canonical_smiles', 'data_validity_comment',
       'data_validity_description', 'document_chembl_id', 'document_journal',
       'document_year', 'ligand_efficiency', 'molecule_chembl_id',
       'molecule_pref_name', 'parent_molecule_chembl_id', 'pchembl_value',
   

This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL3186526&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index([], dtype='object')
This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL3182724&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index([], dtype='object')
This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL3186514&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index([], dtype='object')
This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/ap

This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL66926&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index(['activity_comment', 'activity_id', 'activity_properties',
       'assay_chembl_id', 'assay_description', 'assay_type', 'bao_endpoint',
       'bao_format', 'bao_label', 'canonical_smiles', 'data_validity_comment',
       'data_validity_description', 'document_chembl_id', 'document_journal',
       'document_year', 'ligand_efficiency', 'molecule_chembl_id',
       'molecule_pref_name', 'parent_molecule_chembl_id', 'pchembl_value',
       'potential_duplicate', 'qudt_units', 'record_id', 'relation', 'src_id',
       'standard_flag', 'standard_relation', 'standard_text_value',
       'standard_type', 'standard_units', 'standard_upper_value',
       'standard_value', 'target_chembl_id', 'target_organism',
       'target_pre

This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL486208&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index([], dtype='object')
This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL3586094&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index([], dtype='object')
This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL448502&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index(['activity_comment', 'activity_id', 'activity_properties',
       'assay_chembl_id', 'assay_description', 'assay_type', 'bao_endpoint',
       'ba

This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL1797128&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index([], dtype='object')
This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL273977&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index([], dtype='object')
This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api/data/activity.json?molecule_chembl_id__in=CHEMBL2106930&limit=5&standard_type=IC50

These are the available columns for the Activities API:
Index([], dtype='object')
This is the url string that calls the 'Activities' API with the initial query specification:
https://www.ebi.ac.uk/chembl/api

In [14]:
df.to_csv('till_now_bioactivity_3.csv', index=False)

<font size=4> <b> Scraped Bioactivity Data and saved it as 'till_now_bioactivity_3.csv'.<b> </font>

In [1]:
import pandas as pd
df=pd.read_csv('till_now_bioactivity_3.csv')

In [2]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 13055 entries, 0 to 13054
Data columns (total 11 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   target_chembl_id    13055 non-null  object 
 1   target_organism     9733 non-null   object 
 2   target_pref_name    13055 non-null  object 
 3   molecule_chembl_id  13055 non-null  object 
 4   molecule_pref_name  12845 non-null  object 
 5   pchembl_value       4591 non-null   float64
 6   standard_type       13055 non-null  object 
 7   standard_relation   7283 non-null   object 
 8   standard_value      7364 non-null   float64
 9   standard_units      7329 non-null   object 
 10  assay_chembl_id     13055 non-null  object 
dtypes: float64(2), object(9)
memory usage: 1.1+ MB


<font size=4> <b> Scraped Bioactivity Data and saved it as 'till_now_bioactivity_3.csv'. The standard bioactivity type and units were kept same to maintain uniformity. <b> </font>

In [3]:
df.standard_type.unique()
df_ic50=df.loc[(df['standard_type']=='IC50')&(df['standard_units']=='nM')]
df_ic50
#standard value is the potency of the drug , lower the number better the potency
df_ic50.head()

target_chembl_id        target_organism       target_pref_name  \
0        CHEMBL364  Plasmodium falciparum  Plasmodium falciparum   
1     CHEMBL612545                    NaN              Unchecked   
2    CHEMBL3879801                    NaN     NON-PROTEIN TARGET   
3    CHEMBL2362975                    NaN     No relevant target   
4     CHEMBL612545                    NaN              Unchecked   

  molecule_chembl_id   molecule_pref_name  pchembl_value standard_type  \
0       CHEMBL326602                  NaN            NaN          IC50   
1        CHEMBL37537  PROTOCATECHUIC ACID            NaN          IC50   
2        CHEMBL37537  PROTOCATECHUIC ACID           4.80          IC50   
3        CHEMBL37537  PROTOCATECHUIC ACID           4.57          IC50   
4        CHEMBL37537  PROTOCATECHUIC ACID            NaN          IC50   

  standard_relation  standard_value standard_units assay_chembl_id  
0                 =        800000.0             nM    CHEMBL759677  
1                 >        100000.0             nM    CHEMBL855657  
2                 =         15700.0             nM    CHEMBL855656  
3                 =         27000.0             nM    CHEMBL855658  
4                 =      29500000.0             nM    CHEMBL891104

In [5]:
df_pic=df_ic50.drop(['standard_relation','pchembl_value','assay_chembl_id'], axis=1)
df_pic.head()

target_chembl_id        target_organism       target_pref_name  \
0        CHEMBL364  Plasmodium falciparum  Plasmodium falciparum   
1     CHEMBL612545                    NaN              Unchecked   
2    CHEMBL3879801                    NaN     NON-PROTEIN TARGET   
3    CHEMBL2362975                    NaN     No relevant target   
4     CHEMBL612545                    NaN              Unchecked   

  molecule_chembl_id   molecule_pref_name standard_type  standard_value  \
0       CHEMBL326602                  NaN          IC50        800000.0   
1        CHEMBL37537  PROTOCATECHUIC ACID          IC50        100000.0   
2        CHEMBL37537  PROTOCATECHUIC ACID          IC50         15700.0   
3        CHEMBL37537  PROTOCATECHUIC ACID          IC50         27000.0   
4        CHEMBL37537  PROTOCATECHUIC ACID          IC50      29500000.0   

  standard_units  
0             nM  
1             nM  
2             nM  
3             nM  
4             nM

In [8]:
#df2.info()
df.info()
df_ic50.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 13055 entries, 0 to 13054
Data columns (total 11 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   target_chembl_id    13055 non-null  object 
 1   target_organism     9733 non-null   object 
 2   target_pref_name    13055 non-null  object 
 3   molecule_chembl_id  13055 non-null  object 
 4   molecule_pref_name  12845 non-null  object 
 5   pchembl_value       4591 non-null   float64
 6   standard_type       13055 non-null  object 
 7   standard_relation   7283 non-null   object 
 8   standard_value      7364 non-null   float64
 9   standard_units      7329 non-null   object 
 10  assay_chembl_id     13055 non-null  object 
dtypes: float64(2), object(9)
memory usage: 1.1+ MB
<class 'pandas.core.frame.DataFrame'>
Int64Index: 6553 entries, 0 to 13054
Data columns (total 11 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  ----

In [9]:
df4=df_ic50[df_ic50.standard_value.notna()]
df4.info()
df4.to_csv('bioactivity_data_flavormolecules_actual_ouput22.csv', index=False)

<class 'pandas.core.frame.DataFrame'>
Int64Index: 6553 entries, 0 to 13054
Data columns (total 11 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   target_chembl_id    6553 non-null   object 
 1   target_organism     4735 non-null   object 
 2   target_pref_name    6553 non-null   object 
 3   molecule_chembl_id  6553 non-null   object 
 4   molecule_pref_name  6379 non-null   object 
 5   pchembl_value       4591 non-null   float64
 6   standard_type       6553 non-null   object 
 7   standard_relation   6472 non-null   object 
 8   standard_value      6553 non-null   float64
 9   standard_units      6553 non-null   object 
 10  assay_chembl_id     6553 non-null   object 
dtypes: float64(2), object(9)
memory usage: 614.3+ KB


In [13]:
selection = ['molecule_chembl_id','molecule_pref_name','standard_value','target_chembl_id','target_pref_name','target_organism']
df5 = df4[selection]
rslt_df = df5.sort_values(by = 'standard_value', ascending = True) 
rslt_df.drop_duplicates(subset='molecule_chembl_id', keep='first', inplace=True)
rslt_df.reset_index(drop=True, inplace=True)
rslt_df.info()
rslt_df.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 365 entries, 0 to 364
Data columns (total 6 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   molecule_chembl_id  365 non-null    object 
 1   molecule_pref_name  306 non-null    object 
 2   standard_value      365 non-null    float64
 3   target_chembl_id    365 non-null    object 
 4   target_pref_name    365 non-null    object 
 5   target_organism     293 non-null    object 
dtypes: float64(1), object(5)
memory usage: 17.2+ KB


molecule_chembl_id molecule_pref_name  standard_value target_chembl_id  \
0          CHEMBL165        RESVERATROL          0.0225        CHEMBL399   
1        CHEMBL19224         PAPAVERINE          0.0590     CHEMBL613633   
2           CHEMBL50          QUERCETIN          0.1100    CHEMBL2362975   
3          CHEMBL107         COLCHICINE          0.3600    CHEMBL3879801   
4       CHEMBL441687       GLYCYRRHIZIN          0.4000       CHEMBL3746   

                         target_pref_name    target_organism  
0                                    HeLa       Homo sapiens  
1                                   Ileum  Rattus norvegicus  
2                      No relevant target                NaN  
3                      NON-PROTEIN TARGET                NaN  
4  11-beta-hydroxysteroid dehydrogenase 2       Homo sapiens

In [15]:
rslt_df.to_csv("bioactivity_data_flavormolecules_actual_datapreprocessed_output22.csv", index=False)